addons/mail_client_extension/controllers/main.py

```diff
@@ -182,59 +187,84 @@ def _create_company_from_iap(self, domain):
-        # If multiple are found, take the first one.
-        partner = request.env['res.partner'].search([('email', 'in', [normalized_email, email])], limit=1)
+        if not partner_id:
+
+            # Search for the partner based on the email.
+            # If multiple are found, take the first one.
+            partner = request.env['res.partner'].search([('email', 'in', [normalized_email, email])], limit=1)
+
         if partner:
-            response['partner'] = {
-                'id': partner.id,
-                'name': partner.name,
-                'title': partner.function,
-                'email': partner.email,
-                'image': partner.image_128,
-                'phone': partner.phone,
-                'mobile': partner.mobile,
-                'enrichment_info': None
-            }
-            # if there is already a company for this partner, just take it without enrichment.
-            if partner.parent_id:
-                response['partner']['company'] = self._get_company_dict(partner.parent_id)
+            response['partner'] = self._get_partner_dict(partner)
+            response['partner']['extra_info'] = self._get_partner_extra_info(partner.id)
+            if partner.company_type == 'company':
+                response['partner']['company'] = self._get_company_dict(partner)
             else:
-                company = self._find_existing_company(sender_domain)
-                if not company: # create and enrich company
-                    company, enrichment_info = self._create_company_from_iap(sender_domain)
-                    response['enrichment_info'] = enrichment_info
-                partner.write({'parent_id': company})
-                response['partner']['company'] = self._get_company_dict(company)
-        else: #no partner found
+                if partner.parent_id:
+                    response['partner']['company'] = self._get_company_dict(partner.parent_id)
+                else:
+                    response['partner']['company'] = self._get_company_dict(None)
+        else:  # no partner found
             response['partner'] = {
                 'id': -1,
                 'name': name,
                 'email': email,
-                'enrichment_info': None
+                'enrichment_info': None,
+                'extra_info': self._get_partner_extra_info()
             }
-            company = self._find_existing_company(sender_domain)
+            company = self._find_existing_company(normalized_email)
             if not company:  # create and enrich company
                 company, enrichment_info = self._create_company_from_iap(sender_domain)
                 response['enrichment_info'] = enrichment_info
             response['partner']['company'] = self._get_company_dict(company)
 
         return response
 
+    @http.route('/mail_client_extension/partners/query', type="json", auth="outlook", cors="*")
+    def res_partners_get_by_query(self, query, **kwargs):
```

Comment:
I would do "res_partner_search".

"res_partner" as prefix to comply with existing naming convention.
And then "search" because it's essentially just a search.

Again, a docstring explaining in what context this is used and why we chose to search on those 3 fields would be nice.